pkg/persistence/object_store.go

```diff
@@ -247,6 +250,31 @@ func (s *objectBackupStore) PutBackup(info BackupInfo) error {
 		return kerrors.NewAggregate(errs)
 	}
 
+	// TODO(nrb-csi): only upload these if they're non-nil and/or if EnableCSI is on
```

Comment:
I would like to figure out how to remove the duplication in this function too, but need to look at it some more.

Reply:
I've deduped this except for the special cases at the top by using a map for the keys/contents and looping over those for upload. In my testing, this appears to have continued working as expected.